apps/web/src/hooks/useRoutes.ts

```diff
@@ -0,0 +1,19 @@
+import { useMemo } from 'react';
+import { useLocation } from 'react-router-dom';
+import { PUBLIC_ROUTES_PREFIXES } from '../constants/routes';
+
+export function useRoutes() {
+  const location = useLocation();
+  const inPublicRoute = useMemo(
+    () => !!Array.from(PUBLIC_ROUTES_PREFIXES.values()).find((prefix) => location.pathname.startsWith(prefix)),
+    [location]
+  );
+  const inPrivateRoute = !inPublicRoute;
+  const inStudioRoute = location.pathname.startsWith('/studio');
```

Comment:
<!-- Issues highlight specific problems with the subject under review. These problems can be user-facing or behind the scenes. It is strongly recommended to pair this comment with a suggestion. If you are not sure if a problem exists or not, consider leaving a question.  -->
⚠ **issue:** ‏ Given the foundational nature of this hook and how it's used (in a top-level wrapper), I'm concerned that this will cause the entire component tree to re-render if these are unmemoized ‏

Reply:
Good catch, I added a useMemo.